exes/mqtt-bridge/src/external_to_tfc.cpp

```diff
@@ -0,0 +1,82 @@
+#include <cstdint>
+#include <functional>
+#include <string>
+#include <tuple>
+#include <type_traits>
+#include <variant>
+
+#include <boost/asio.hpp>
+#include <tfc/ipc.hpp>
+
+#include "external_to_tfc.hpp"
+#include "spark_plug_interface.hpp"
+
+namespace tfc::mqtt {
+
+namespace asio = boost::asio;
+
+external_to_tfc::external_to_tfc(asio::io_context& io_ctx, tfc::mqtt::spark_plug_interface& spark_plug)
+    : io_ctx_(io_ctx), spark_plug_(spark_plug) {
+  create_outward_signals();
+}
+
+auto external_to_tfc::create_outward_signals() -> void {
+  for (auto const& sig : config_.value().writeable_signals) {
+    if (!sig.name.empty()) {
+      switch (sig.type) {
+        using enum tfc::ipc::details::type_e;
+        case _bool: {
+          outward_signals_.emplace_back(tfc::ipc::bool_signal{ io_ctx_, ipc_client_, sig.name, sig.description });
+          break;
+        }
+        case _double_t: {
+          outward_signals_.emplace_back(tfc::ipc::double_signal{ io_ctx_, ipc_client_, sig.name, sig.description });
+          break;
+        }
+        case _int64_t: {
+          outward_signals_.emplace_back(tfc::ipc::int_signal{ io_ctx_, ipc_client_, sig.name, sig.description });
+          break;
+        }
+        case _json: {
+          outward_signals_.emplace_back(tfc::ipc::json_signal{ io_ctx_, ipc_client_, sig.name, sig.description });
+          break;
+        }
+        case _string: {
+          outward_signals_.emplace_back(tfc::ipc::string_signal{ io_ctx_, ipc_client_, sig.name, sig.description });
+          break;
+        }
+        case _uint64_t: {
+          outward_signals_.emplace_back(tfc::ipc::uint_signal{ io_ctx_, ipc_client_, sig.name, sig.description });
+          break;
+        }
+        default: {
+          logger_.error("Unknown type for signal: {}", sig.name);
+        }
```

Comment:
how about using make_any_signal in ipc?

Reply:
implemented, function reduced to 7 lines.